lib/src/DynamicProto.ts

```diff
@@ -414,13 +414,21 @@ function _populatePrototype(proto:any, className:string, target:any, baseInstFun
  * @ignore
  */
 function _checkPrototype(classProto:any, thisTarget:any) {
-    let thisProto = _getObjProto(thisTarget);
-    while (thisProto && !_isObjectArrayOrFunctionPrototype(thisProto)) {
-        if (thisProto === classProto) {
-            return true;
-        }
+    // This method doesn't existing in older browsers (e.g. IE8)
+    if (_objGetPrototypeOf) {
+        // As this is primarily a coding time check, don't bother checking if running in IE8 or lower
+        let visited:any[] = [];
```

Comment:
Copied the visited workaround (to avoid the endless loop) from the previous fix from March -- which fixed IE8 support